src/api/lib/backend/api/build_results/binaries.rb

```diff
@@ -64,16 +64,14 @@ def self.wipe_published_locked(project_name)
         # the name of the binary as keys and the architectures as the value
         def self.transform_binary_packages_response(response)
           list = {}
+          list.default = []
           parsed_response = Xmlhash.parse(response)
           return list if parsed_response.blank?
           packages = [parsed_response["packages"]].flatten
           packages.each do |build|
             architectures = [build["arch"]].flatten
```

Comment:
[nitpick] architecture_names instead of architectures

Reply:
ok, let's refactor everything :rofl: